webapp/src/blocks.tsx

```diff
@@ -1108,6 +1110,17 @@ export class Editor extends toolboxeditor.ToolboxEditor {
         return this.filterBlocks(subns, this.nsMap[ns]) || [];
     }
 
+    // TODO(jb): Generalize to snippet builder and pass in specific config.
+    private openSnippetDialog() {
+        if (!this.snippetDialog) {
+            const wrapper = document.body.appendChild(document.createElement('div'));
+            const props = { parent: this.parent, mainWorkspace: this.editor }
+            this.snippetDialog = ReactDOM.render(
```

Comment:
that's not really how we add new "dialogs" in the react tree.

Reply:
The only example I could find of a sidebar button opening a dialog was the create function and followed that path. Do you know of a better example somewhere in the codebase?